internal/consensus/reactor.go

```diff
@@ -638,7 +638,7 @@ OUTER_LOOP:
 // pickSendVote picks a vote and sends it to the peer. It will return true if
 // there is a vote to send and false otherwise.
 func (r *Reactor) pickSendVote(ps *PeerState, votes types.VoteSetReader) bool {
-	if vote, ok := ps.PickVoteToSend(votes); ok {
+	if vote, ok := ps.PickVoteToSend(votes); ok && vote != nil {
```

Comment:
It seems odd that this method would return `ok == true` and a nil vote. Is that a bug in the implementation, or does that condition actually make sense?

Reply:
Yeah I could also adjust `ps.PickVoteSend` so that it will return false if it can't find the vote. That might be a better way of doing things instead of adding this extra clause